_studio/mfx_lib/shared/src/mfx_vc1_dec_common.cpp

```diff
@@ -121,7 +121,7 @@ mfxStatus MFXVC1DecCommon::Query(VideoCORE* core, mfxVideoParam *in, mfxVideoPar
                 (in->IOPattern & MFX_IOPATTERN_OUT_OPAQUE_MEMORY))
             {
                 uint32_t mask = in->IOPattern & 0xf0;
-                if (mask != MFX_IOPATTERN_OUT_VIDEO_MEMORY || mask != MFX_IOPATTERN_OUT_SYSTEM_MEMORY || mask != MFX_IOPATTERN_OUT_OPAQUE_MEMORY)
+                if (mask != MFX_IOPATTERN_OUT_VIDEO_MEMORY && mask != MFX_IOPATTERN_OUT_SYSTEM_MEMORY && mask != MFX_IOPATTERN_OUT_OPAQUE_MEMORY)
```

Comment:
Logic is changed here. Is it done on purpose ?

Reply:
I took a closer look to that. This condition on line 120 seems to be not needed, because there's another similar condition after on line 182.
```cpp
        if ((in->IOPattern & MFX_IOPATTERN_OUT_SYSTEM_MEMORY) ||
            (in->IOPattern & MFX_IOPATTERN_OUT_VIDEO_MEMORY) ||
            (in->IOPattern & MFX_IOPATTERN_OUT_OPAQUE_MEMORY))
            out->IOPattern = in->IOPattern;
```
And it overrides results of this assignment to `out->IOPattern` anyway and `in->IOPattern` value is not changed and `out->IOPattern` value is not used between this section and line 182.
So I'm going to remove this condition.